### PR TITLE
chore(apprunner): add validation cpu and memory for app runner

### LINKED ISSUE #25872: @aws-cdk_aws-apprunner-alpha: Invalid cpu or memory should be validated
### Describe the feature

App Runner's CPU and Memory have a fixed input patterns in [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-instanceconfiguration.html#cfn-apprunner-service-instanceconfiguration-cpu).
However, in CDK's L2 Construct (alpha), the input format is free (like `Some vCPU`). Tests are as follows (packages/@aws-cdk/aws-apprunner-alpha/test/service.test.ts).

```ts
test('custom cpu and memory units are allowed', () => {
  // GIVEN
  const app = new cdk.App();
  const stack = new cdk.Stack(app, 'demo-stack');
  // WHEN
  new apprunner.Service(stack, 'DemoService', {
    source: apprunner.Source.fromEcrPublic({
      imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
    }),
    cpu: apprunner.Cpu.of('Some vCPU'),
    memory: apprunner.Memory.of('Some GB'),
  });
  // THEN
  Template.fromStack(stack).hasResourceProperties('AWS::AppRunner::Service', {
    InstanceConfiguration: {
      Cpu: 'Some vCPU',
      Memory: 'Some GB',
    },
    NetworkConfiguration: {
      EgressConfiguration: {
        EgressType: 'DEFAULT',
      },
    },
  });
});
```

I would like to validate an invalid format before deploy.

### Use Case

Incorrect values can be detected at the synth stage before deployment.

### Proposed Solution

Create a private method for validation and call it at the constructor in the `Service` class.

### Other Information

CFn output is the following message when deploy errors.

`Properties validation failed for resource AppRunnerConstructAppRunnerService123456789 with message: #/InstanceConfiguration/Cpu: failed validation constraint for keyword [pattern]`

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

v2.82.0

### Environment details (OS name and version, etc.)

MacOS Ventura 13.2

### PR DESCRIPTION
**Pull request by go-to-k**
This PR adds the same validation for the App Runner's CPU and Memory values as [CloudFormation's input patterns](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-instanceconfiguration.html#cfn-apprunner-service-instanceconfiguration-cpu).

Closes #25872

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by go-to-k** [comment]
I revert [this commit](https://github.com/aws/aws-cdk/pull/25877/commits/7bf506c563af22c11174a3ef279b37376528a055) for the following reasons.

- Make it easier to understand even if it is somewhat redundant.
- Less mistakes when dealing with new specs.
  - e.g. even in the case of adding only the enum option and forgetting to add it to the `of()` validation parameter, the test can still detect the error.
- Even static variable definitions are secured by defining them through a safe `of()` with validation.

**Comment by go-to-k** [comment]
The AWS CodeBuild CI Report failed with internal health check errors...

`time="2023-06-07T09:58:28.630630240Z" level=error msg="healthcheck failed fatally"`

https://github.com/aws/aws-cdk/pull/25877#issuecomment-1580349863

**Review by otaviomacedo — DISMISSED** [review]

File: packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ export enum ImageRepositoryType {
   ECR = 'ECR',
 }
 
+/**
+ * CPU input patterns for App Runner service
+ */
+export type CpuPatternType = '256' | '512' | '1024' | '2048' | '4096' | '0.25 vCPU' | '0.5 vCPU' | '1 vCPU' | '2 vCPU' | '4 vCPU'
```
[the remark this answers is not in the document]
Comment (go-to-k) [reply packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts]: @otaviomacedo 

Thanks for your review. I reverted to the previous commit.

So please review again.

File: packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts [hunk 2]
```diff
@@ -61,15 +66,20 @@ export class Cpu {
...
    * @param unit The unit of CPU.
    */
-  private constructor(public readonly unit: string) {}
+  private constructor(public readonly unit: CpuPatternType) {}
 }
 
+/**
+ * Memory input patterns for App Runner service
+ */
+export type MemoryPatternType = '512' | '1024' | '2048' | '3072' | '4096' | '6144' | '8192' | '10240' | '12288' | '0.5 GB' | '1 GB' | '2 GB' | '3 GB' | '4 GB' | '6 GB' | '8 GB' | '10 GB' | '12 GB';
```
[the remark this answers is not in the document]
Comment (go-to-k) [reply packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts]: I reverted same as above comments.

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**